File: web/src/app/messages/message.ts
// A line of the inbox or of the sent folder, as the API lists it. Mine
// is true for a message I sent; readAt is null until the recipient opens
// it.
export interface Message {
  id: number;
  from: string;
  to: string;
  subject: string;
  sentAt: string;
  readAt: string | null;
  mine: boolean;
}

export interface MessageDetail extends Message {
  body: string;
}

// What a member sends. The sender is never in it: the API knows me from
// the token.
export interface MessageDraft {
  to: string;
  subject: string;
  body: string;
}

File: api/src/HomeMarket.Api/Models/Message.cs
namespace HomeMarket.Api.Models
{
    // A note from one account to another. ReadAt stays null until the
    // recipient opens it.
    public class Message
    {
        public int Id { get; set; }
        public string Subject { get; set; } = string.Empty;
        public string Body { get; set; } = string.Empty;
        public DateTime SentAt { get; set; }
        public DateTime? ReadAt { get; set; }
        public int SenderId { get; set; }
        public Account? Sender { get; set; }
        public int RecipientId { get; set; }
        public Account? Recipient { get; set; }
    }
}
